torch/distributed/fsdp/_common_utils.py

```diff
@@ -177,6 +177,10 @@ def _module_handles(state: _FSDPState, module: nn.Module) -> List:
     the handles that contain some parameter in ``module``.
     """
     if _is_composable(state):
+        # A valid FSDP state may have no managed parameters and hence no
+        # handles, meaning no entry in `_fully_sharded_module_to_handles`
+        if len(state._handles) == 0:
+            return []
```

Comment:
I think the `_has_fsdp_params()` check is still valid. Before, the composable path would raise an error when it did not need to, which is why I had to add this case.

In other words, this is covered by the existing tests. `summon_full_params()` on a module with `fully_shard()` applied but no managed parameters would error otherwise.